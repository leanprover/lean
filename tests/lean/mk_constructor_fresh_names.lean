open tactic

example (fst fst_1 : nat) : fst = fst :=
by do
  ns ← mk_constructors_fresh_names `prod,
  trace ns, -- [[fst_2, snd]]
  constructor

example (a : nat) : a = a :=
by do
  ns ← mk_constructors_fresh_names `acc,
  trace ns, -- [[x, a_1]
  constructor

inductive Foo
| mk₁ (a b c : nat)  : Foo
| mk₂ (d e : bool)   : Foo
| mk₃ (f g : Foo)    : Foo

example (a d d_1 e : bool) : a = a :=
by do
  ns ← mk_constructors_fresh_names `Foo,
  trace ns, -- [[a_1, b, c], [d_2, e_1], [f, g]]
  constructor
