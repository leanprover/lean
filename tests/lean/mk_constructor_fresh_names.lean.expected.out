[[fst_2, snd]]
[[x, _x]]
[[a_1, b, c], [d_2, e_1], [f, g]]
